airflow/executors/kubernetes_executor.py

```diff
@@ -370,7 +416,17 @@ def sync(self) -> None:
         """
         self.log.debug("Syncing KubernetesExecutor")
         self._health_check_kube_watcher()
-        while True:
+
+        multi_threads_queue_process(
+            queue_size=self.watcher_queue.qsize(),
+            queue_type='watcher',
+            process_method=self.process_watcher_queue,
```

Comment:
what is the point of multithreading `process_watcher_queue`?

it does not appear that it makes any network calls.

Reply:
@Dinghang

Reply:
@Dinghang still working on this would love to get it merged

Reply:
Hi @dstandish , sorry for the late reply. Was AFK due to some personal issues. Just got back. Will be addressing comments accordingly. Gonna ping you again once it's ready.